### PR TITLE
Frontend Port Change and Error Handling Improvements

### PR DESCRIPTION
**Pull request by thiagobardini**
**Summary:** This pull request includes changes to the frontend port, the addition of a custom Loader, and isError handling for missing data in HomeTable and UserTable.

- [changed the frontend port to 3001](https://github.com/codeforboston/urban-league-heat-pump-accelerator/commit/8314db7182e68efe6331eea0d7fafecb44c9e496)
- [Created custom Loader; kept previous version in case we want to use it again](https://github.com/codeforboston/urban-league-heat-pump-accelerator/commit/8a00a2dc081d6a9ee1cea56d1c724dbedd26e6aa)
- [Added isError handling for missing data in HomeTable](https://github.com/codeforboston/urban-league-heat-pump-accelerator/commit/c8e41d8211742437a3ba95b85423d52cdb1ee3eb)
- [Added isError handling for missing data in UserTable](https://github.com/codeforboston/urban-league-heat-pump-accelerator/commit/bcf73ae09e11a6c749e5df3fe4aaf05c25195244)

### CODE REVIEW

File: frontend/front/src/components/Loader.js
Comment (LWest001) [inline frontend/front/src/components/Loader.js:1]: New loader! Very cool.

File: frontend/front/src/pages/Admin/home/HomeTable.js
Comment (LWest001) [inline frontend/front/src/pages/Admin/home/HomeTable.js:1]: Might be good to have a unique message for this page: "Error fetching homes data"

File: frontend/front/src/pages/Admin/user/UserTable.js
Comment (LWest001) [inline frontend/front/src/pages/Admin/user/UserTable.js:1]: Same thing here: ""Error fetching surveyors data"

File: frontend/front/src/pages/Admin/home/HomeTable.js
Comment (LWest001) [inline frontend/front/src/pages/Admin/home/HomeTable.js:1]: Might be good to have a unique message for this page: "Error fetching homes data"

**Review by LWest001 — CHANGES_REQUESTED** [review]
Looks good, all functioning properly. New loader has a lot more personality. I think the error messages should be specific to the type of data being loaded.

**Comment by thiagobardini** [comment]
@LWest001, if you approve the changes, let's merge this PR, and I'll create a new PR for my other ticket.

File: frontend/front/package.json [hunk 1]
```diff
@@ -28,7 +28,7 @@
     "web-vitals": "^2.1.0"
   },
   "scripts": {
-    "start": "react-scripts start",
+    "start": "PORT=3001 react-scripts start",
```
Comment (mattdelsordo) [inline frontend/front/package.json:31]: I'm out of the loop, is this just so it'll play nicer with a local back end?

File: frontend/front/src/components/PreviousLoader.js
Comment (mattdelsordo) [inline frontend/front/src/components/PreviousLoader.js:1]: Do we really need to keep this? IMO we should just stick to the new one everywhere.

File: frontend/front/src/components/SurveyComponent/SurveyComponent.js [hunk 2]
```diff
@@ -259,22 +265,19 @@ const SurveyComponentWrapper = forwardRef((props, ref) => {
     return null;
   }, [defaultData, surveyStructure]);
 
+  if (isSurveyLoading) {
+    return <Loader />;
+  }
   return (
     <div ref={ref} style={style}>
-      {surveyStructure && formDefault && activeHome ? (
+      {!surveyStructure && formDefault && activeHome ? (
```
Comment (mattdelsordo) [inline frontend/front/src/components/SurveyComponent/SurveyComponent.js:273]: I think this logic is wrong, `surveyStructure` needs to exist for the `SurveyComponent` to render.

**Review by mattdelsordo — CHANGES_REQUESTED** [review]

File: frontend/front/package.json (hunk 1 above)
Comment (LWest001) [reply frontend/front/package.json:31]: That's right - the backend runs on port 3000.

File: frontend/front/src/components/SurveyComponent/SurveyComponent.js (hunk 2 above)
Comment (LWest001) [reply frontend/front/src/components/SurveyComponent/SurveyComponent.js:273]: I agree, I think the previous logic was correct.

**Review by LWest001 — CHANGES_REQUESTED** [review]
Reformatted error messages look good. Check comments on `SurveyComponent.js`

File: frontend/front/src/components/PreviousLoader.js
Comment (thiagobardini) [reply frontend/front/src/components/PreviousLoader.js:1]: We don't need to keep this. I can remove it. I just want to make sure if you guys like the new version. Should I remove it?

File: frontend/front/src/components/SurveyComponent/SurveyComponent.js (hunk 2 above)
Comment (thiagobardini) [reply frontend/front/src/components/SurveyComponent/SurveyComponent.js:273]: Sorry, I did that to test the error message. I will fix it. Thank you!

File: frontend/front/src/components/SurveyComponent/SurveyComponent.js [hunk 3]
```diff
@@ -31,6 +33,7 @@ const SurveyComponent = ({
   formDefault,
   surveyStructure,
   onDelete,
+  isErrorSurvey,
```
Comment (LWest001) [inline frontend/front/src/components/SurveyComponent/SurveyComponent.js:36]: This unused prop should be removed - `isSurveyError` (line 252) serves this purpose.

**Review by LWest001 — CHANGES_REQUESTED** [review]
1 comment, then this should be good for a merge.

File: frontend/front/src/components/PreviousLoader.js
Comment (LWest001) [reply frontend/front/src/components/PreviousLoader.js:1]: I think we should remove it. There are also still some of the basic `<CircularProgress/>` loaders floating around the app, so we can make another issue to replace them with the new `<Loader/>`.

Comment (mattdelsordo) [reply frontend/front/src/components/PreviousLoader.js:1]: Agreed

File: frontend/front/src/components/SurveyComponent/SurveyComponent.js (hunk 3 above)
Comment (thiagobardini) [reply frontend/front/src/components/SurveyComponent/SurveyComponent.js:36]: I saw some examples using isError. This is a boolean, but I'll change it to error. Thanks!

**Review by LWest001 — APPROVED** [review]
Looks good!

**Review by mattdelsordo — APPROVED** [review]

**Pull request merged**